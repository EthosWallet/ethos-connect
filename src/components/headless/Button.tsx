import Loader from '../svg/Loader'
import React from 'react'

const Button = (props: any) => {
  const { children, onClick, isWorking, workingComponent, ...reactProps } = props

  return (
    <button onClick={onClick} {...reactProps}>
      {isWorking ? (
        workingComponent || (
          <span className="block p-2">
            <Loader width={30} />
          </span>
        )
      ) : (
        <>{children}</>
      )}
    </button>
  )
}
export default Button
