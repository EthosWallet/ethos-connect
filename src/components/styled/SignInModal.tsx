import React, { useState } from 'react'
import login from '../../lib/login'
import WalletConnect from '../svg/WalletConnect'
import Ethos from '../svg/Ethos'
import Metamask from '../svg/Metamask'
import Google from '../svg/Google'
import Loader from '../svg/Loader'
import getConfiguration from '../../lib/getConfiguration'
// import { useConnect } from 'wagmi'
// import { Chain } from '../../enums/Chain'
import Sui from '../svg/Sui'
import useWindowDimensions from '../../hooks/useWindowDimensions'
import { Breakpoints } from '../../enums/Breakpoints'
import connectSui from '../../lib/connectSui'
import event from '../../lib/event'

export type SignInModalProps = {
  isOpen: boolean
  onClose?: () => void
  onEmailSent?: () => void
}

const SignInModal = ({ isOpen, onClose, onEmailSent }: SignInModalProps) => {
  const { width } = useWindowDimensions()

  const { appId, walletAppUrl } = getConfiguration()
  // const eth = chain === Chain.Eth

  const [showMissingMessage, setShowMissingMessage] = useState<boolean>(false)
  const [signingIn, setSigningIn] = useState(false)
  const [email, setEmail] = useState('')

  // const { connect, connectors, error, isLoading, pendingConnector } = eth
  //   ? useConnect()
  //   : {
  //       connect: null,
  //       connectors: [],
  //       error: null,
  //       isLoading: false,
  //       pendingConnector: null,
  //     }

  const sendEmail = async () => {
    setSigningIn(true)
    await login({ email, appId })
    setEmail('')
    onEmailSent && onEmailSent()
    onClose && onClose()
    setSigningIn(false)
    event({ action: 'send_email', category: 'sign_in', label: email, value: 1})
  }

  const loginWithGoogle = () => {
    login({ provider: 'google', appId })
  }

  const connectEthos = () => {
    setShowMissingMessage(true)
  }

  const _connectSui = async () => {
    const connected = await connectSui()
    if (!connected) {
      setShowMissingMessage(true)
    } else {
      onClose && onClose()
    }
    event({ action: 'connect', category: 'sign_in', label: 'sui', value: connected ? 1: 0})
  }

  const logo = (connectorId: string) => {
    switch (connectorId) {
      case 'metaMask':
        return <Metamask />
      // case 'Coinbase Wallet':
      case 'ethos':
        return <Ethos width={17} />
      case 'sui':
        return <Sui width={15} />
      default:
        return <WalletConnect />
    }
  }

  return (
    <div style={dialogStyle(isOpen)} role="dialog">
      <div style={backdropStyle()} onClick={() => console.log('clicked')} />

      <div style={modalOuterWrapperStyle()}>
        <div style={modalInnerWrapperStyle(width)}>
          <div style={dialogPanelStyle(width)}>
            <div>
              <div style={headerStyle()}>
                <h3 style={titleStyle()}>Sign In</h3>
                <div style={closeStyle()} onClick={onClose}>
                  &#x2715;
                </div>
              </div>
              <div style={mainContentStyle(width)}>
                <div style={walletOptionsStyle(width)}>
                  <div style={walletOptionStyle()} onClick={connectEthos}>
                    <button style={walletOptionButtonStyle()}>
                      {logo('ethos')}
                      Ethos
                    </button>
                  </div>
                  <div style={walletOptionStyle()} onClick={_connectSui}>
                    <button style={walletOptionButtonStyle()}>
                      {logo('sui')}
                      Sui Test Wallet
                    </button>
                  </div>
                  {/* {isOpen &&
                    connectors.map((connector: any) => (
                      <div
                        key={connector.id}
                        style={walletOptionStyle()}
                        onClick={() => connect!({ connector })}
                      >
                        <button disabled={!connector.ready} style={walletOptionButtonStyle()}>
                          {logo(connector.id)}
                          {connector.name}
                          {!connector.ready && (
                            <span style={connectorSubStyle()}>(unsupported)</span>
                          )}
                          {isLoading && pendingConnector?.id === connector.id && (
                            <span style={connectorSubStyle()}>(connecting)</span>
                          )}
                        </button>
                      </div>
                    ))}
                  {error && <div style={connectorWarning()}>{error.message}</div>} */}
                  {showMissingMessage && (
                    <div style={connectorWarning()}>
                      You do not have the necessary wallet extension installed.
                    </div>
                  )}
                </div>
                <div style={registrationStyle(width)}>
                  <div>
                    <h3 style={registrationHeaderStyle()}>
                      Sign up or log in with Google
                    </h3>
                    <div style={socialLoginButtonsStyle()} onClick={loginWithGoogle}>
                      <div style={socialLoginButtonStyle()}>
                        <Google width={36} />
                      </div>
                    </div>
                  </div>
                  <h3 style={registrationHeaderStyle()}>
                    Or sign up or log in with a link
                  </h3>
                  {signingIn ? (
                    <div style={loaderStyle()}>
                      <Loader width={50} />
                    </div>
                  ) : (
                    <>
                      <form onSubmit={sendEmail}>
                        <input
                          style={inputStyle()}
                          type="email"
                          placeholder="Email address"
                          value={email}
                          onChange={(e) => setEmail(e.target.value)}
                        />
                        <button style={buttonStyle(width)} type="submit">
                          Send
                        </button>
                      </form>
                      <div style={selfCustodialSection()}>
                        Advanced:&nbsp;
                        <a
                          href={`${walletAppUrl}/self-custodial`}
                          style={selfCustodialLink()}
                          target="_blank"
                          rel="noopener noreferrer"
                        >
                          Create A Self-Custodial Wallet
                        </a>
                      </div>
                    </>
                  )}
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  )
}

/*
How to convert tailwind to inline CSS:
Paste the tailwind styless into https://tailwind-to-css.vercel.app/
Set aside the media queries
Paste that output into https://staxmanade.com/CssToReact/
Add media queries using `modalInnerWrapperStyle` as an example
*/

const dialogStyle = (isOpen: boolean) =>
  ({
    display: isOpen ? 'block' : 'none',
    position: 'relative',
    zIndex: '10',
  } as React.CSSProperties)

// const mainWrapper = (isOpen: boolean) =>
//   // flex justify-center items-center
//   ({
//     display: isOpen ? 'block' : 'none',
//     justifyContent: 'center',
//     alignItems: 'center',
//   } as React.CSSProperties)

const backdropStyle = () =>
  // fixed inset-0 bg-gray-500 bg-opacity-75 transition-opacity
  ({
    position: 'fixed',
    top: '0px',
    right: '0px',
    bottom: '0px',
    left: '0px',
    backgroundColor: 'rgb(107 114 128 / .75)',
  } as React.CSSProperties)

const modalOuterWrapperStyle = () =>
  // fixed z-10 inset-0 overflow-y-auto
  ({
    position: 'fixed',
    zIndex: '99',
    top: '0px',
    right: '0px',
    bottom: '0px',
    left: '0px',
    overflowY: 'auto',
  } as React.CSSProperties)

const modalInnerWrapperStyle = (width: number): React.CSSProperties => {
  // flex items-end sm:items-center justify-center min-h-full p-4 text-center sm:p-0
  const styles = {
    display: 'flex',
    alignItems: 'flex-end',
    justifyContent: 'center',
    minHeight: '100%',
    padding: '1rem' /* 16px */,
    textAlign: 'center',
  }
  const sm = {
    padding: '0',
    alignItems: 'center',
  }

  return width < Breakpoints.sm
    ? (styles as React.CSSProperties)
    : ({ ...styles, ...sm } as React.CSSProperties)
}

const dialogPanelStyle = (width: number) => {
  // relative bg-white rounded-lg px-4 pt-5 pb-4 text-left overflow-hidden shadow-xl transform transition-all sm:my-8 sm:max-w-lg sm:w-full sm:p-6
  const styles = {
    overflow: 'hidden',
    position: 'relative',
    backgroundColor: '#ffffff',
    transitionProperty: 'all',
    textAlign: 'left',
    borderRadius: '0.5rem',
    boxShadow: '0 20px 25px -5px rgba(0, 0, 0, 0.1), 0 10px 10px -5px rgba(0, 0, 0, 0.04)',
  }
  const sm = {
    marginTop: '2rem',
    marginBottom: '2rem',
    width: '100%',
    maxWidth: '40rem',
  }

  return width < Breakpoints.sm
    ? (styles as React.CSSProperties)
    : ({ ...styles, ...sm } as React.CSSProperties)
}

const closeStyle = () =>
  ({
    backgroundColor: '#F9FAFB',
    borderRadius: '100%',
    width: '24px',
    height: '24px',
    textAlign: 'center',
    color: '#A0AEBA',
    cursor: 'pointer',
  } as React.CSSProperties)

// const modalStyle = (isOpen: boolean) =>
//   ({
//     textAlign: 'left',
//     border: '1px solid rgb(203 213 225)',
//     borderRadius: '0.5rem',
//     transitionProperty: 'opacity',
//     transitionTimingFunction: 'cubic-bezier(0.4, 0, 0.2, 1)',
//     transitionDuration: '250ms',
//     opacity: isOpen ? 1 : 0,
//     position: 'absolute',
//     left: isOpen ? '50%' : '-9999px',
//     top: '40%',
//     transform: 'translate(-50%, -50%)',
//     backgroundColor: 'white',
//     width: '660px',
//     fontWeight: '400',
//     zIndex: '99',
//   } as React.CSSProperties)

const headerStyle = () =>
  ({
    borderBottom: '1px solid rgb(241 245 249)',
    padding: '12px',
    display: 'flex',
    justifyContent: 'space-between',
  } as React.CSSProperties)

const titleStyle = () =>
  ({
    fontSize: '1rem',
    fontWeight: '500',
    margin: '0',
  } as React.CSSProperties)

const mainContentStyle = (width: number) => {
  const styles = {
    justifyContent: 'space-between',
  }
  const sm = {
    display: 'flex',
  }

  return width < Breakpoints.sm
    ? (styles as React.CSSProperties)
    : ({ ...styles, ...sm } as React.CSSProperties)
}

const walletOptionsStyle = (width: number) => {
  const styles = {
    padding: '18px',
    gap: '6px',
    display: 'flex',
    flexDirection: 'column',
    borderBottom: '1px solid rgb(241 245 249)',
  }
  const sm = {
    width: '300px',
    padding: '24px 12px',
    borderRight: '1px solid rgb(241 245 249)',
  }

  return width < Breakpoints.sm
    ? (styles as React.CSSProperties)
    : ({ ...styles, ...sm } as React.CSSProperties)
}

const walletOptionStyle = () =>
  ({
    padding: '12px',
    backgroundColor: '#F9FAFB',
    borderRadius: '0.5rem',
    fontWeight: '500',
    cursor: 'pointer',
  } as React.CSSProperties)

const walletOptionButtonStyle = () =>
  ({
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'start',
    gap: '0.5rem',
    border: 'none',
    background: 'none',
    textDecoration: 'none',
  } as React.CSSProperties)

// const connectorSubStyle = () => ({
//   fontWeight: '300',
//   color: 'gray',
//   fontSize: 'smaller',
// })

const socialLoginButtonsStyle = () => ({
  padding: '12px 0',
  display: 'flex',
  gap: '6px'
})

const socialLoginButtonStyle = () => ({
  cursor: 'pointer'
})

const registrationStyle = (width: number) => {
  const styles = {
    padding: '18px',
    display: 'flex',
    flexDirection: 'column',
    gap: '12px',
  }
  const sm = {
    padding: '24px',
  }

  return width < Breakpoints.sm
    ? (styles as React.CSSProperties)
    : ({ ...styles, ...sm } as React.CSSProperties)
}

const registrationHeaderStyle = () =>
  ({
    fontWeight: '500',
    margin: '0',
  } as React.CSSProperties)

// const explainerStyle = () =>
//   ({
//     fontSize: 'smaller',
//   } as React.CSSProperties)

const inputStyle = () =>
  ({
    border: '1px solid rgb(203 213 225)',
    borderRadius: '0.5rem',
    padding: '6px',
    width: '90%',
  } as React.CSSProperties)

const buttonStyle = (width: number) => {
  const styles = {
    marginTop: '0.5rem',
    border: '1px solid rgb(203 213 225)',
    borderRadius: '0.5rem',
    padding: '6px',
    backgroundColor: '#761AC7',
    color: '#FFFFFF',
    textDecoration: 'none',
    minWidth: '6rem',
  }
  const sm = {
    marginTop: '1rem',
  }

  return width < Breakpoints.sm
    ? (styles as React.CSSProperties)
    : ({ ...styles, ...sm } as React.CSSProperties)
}

const loaderStyle = () =>
  ({
    display: 'flex',
    justifyContent: 'center',
    padding: '45px 0',
  } as React.CSSProperties)

const connectorWarning = () =>
  ({
    fontSize: 'small',
    textAlign: 'center',
    paddingTop: '6px',
  } as React.CSSProperties)

const selfCustodialSection = () =>
  ({
    paddingTop: '6px',
    fontSize: 'small',
    paddingLeft: '3px',
  } as React.CSSProperties)

const selfCustodialLink = () =>
  ({
    color: '#751ac7',
    textDecoration: 'underline',
    fontWeight: 400,
  } as React.CSSProperties)

export default SignInModal
