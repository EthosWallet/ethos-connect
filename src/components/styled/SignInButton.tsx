import React from 'react'
import Button from '../headless/Button'
import SignInModal from './SignInModal'

export interface SignInButtonProps extends React.HTMLAttributes<HTMLButtonElement> {
  onClick?: () => void
  onLoaded?: () => void
  onEmailSent?: () => void
}

const SignInButton = (props: SignInButtonProps) => {
  const { children, onClick, onEmailSent, ...reactProps } = props

  const [isOpen, setIsOpen] = React.useState(false)

  const _onClick = () => {
    setIsOpen(true)

    onClick && onClick()
  }

  return (
    <>
      <SignInModal isOpen={isOpen} onEmailSent={onEmailSent} onClose={() => setIsOpen(false)} />
      <Button onClick={_onClick} {...reactProps}>
        {children || <>Sign In</>}
      </Button>
    </>
  )
}
export default SignInButton
