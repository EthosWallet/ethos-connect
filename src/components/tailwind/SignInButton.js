import React from 'react';
import Button from '../headless/Button';
import SignInModal from './SignInModal';

const SignInButton = (props) => {
  const { appId, children, onClick, onSignIn, ...reactProps } = props;

  const [isOpen, setIsOpen] = React.useState(false);

  const _onClick = (e) => {
    setIsOpen(true);

    if (onClick) {
      onClick(e);
    }
  }

  return (
    <>
      <SignInModal 
        appId={appId}
        isOpen={isOpen} 
        onSignIn={onSignIn}
        onClose={() => setIsOpen(false)} 
      />
      <Button
        onClick={_onClick}
        isWorking={isOpen}
        {...reactProps}
      >
        {props.children || <>Sign In</>}
      </Button>

    </>
  )
}
export default SignInButton;