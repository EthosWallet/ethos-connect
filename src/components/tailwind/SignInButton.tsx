import React from 'react'
import Button from '../headless/Button'
import SignInModal from './SignInModal'
import { WagmiConfig, createClient, defaultChains, configureChains } from 'wagmi'

import { alchemyProvider } from 'wagmi/providers/alchemy'
import { publicProvider } from 'wagmi/providers/public'

// import { CoinbaseWalletConnector } from 'wagmi/connectors/coinbaseWallet'
// import { InjectedConnector } from 'wagmi/connectors/injected'
import { MetaMaskConnector } from 'wagmi/connectors/metaMask'
import { WalletConnectConnector } from 'wagmi/connectors/walletConnect'

const { chains, provider, webSocketProvider } = configureChains(defaultChains, [
  alchemyProvider({ alchemyId: 'rFuRqqP4t7WQ0hFHrN6h7HdVXOatTnBV' }),
  publicProvider(),
])

// Set up client
const client = createClient({
  autoConnect: true,
  connectors: [
    new MetaMaskConnector({ chains }),
    // new CoinbaseWalletConnector({
    //   chains,
    //   options: {
    //     appName: 'wagmi',
    //   },
    // }),
    new WalletConnectConnector({
      chains,
      options: {
        qrcode: true,
      },
    }),
    // new InjectedConnector({
    //   chains,
    //   options: {
    //     name: 'Injected',
    //     shimDisconnect: true,
    //   },
    // }),
  ],
  provider,
  webSocketProvider,
})

const SignInButton = (props: any) => {
  const { appId, children, onClick, onEmailSent, onProviderSelected, ...reactProps } = props

  const [isOpen, setIsOpen] = React.useState(false)

  const _onClick = (e: any) => {
    setIsOpen(true)

    if (onClick) {
      onClick(e)
    }
  }

  return (
    <WagmiConfig client={client}>
      <SignInModal
        isOpen={isOpen}
        onEmailSent={onEmailSent}
        onProviderSelected={onProviderSelected}
        onClose={() => setIsOpen(false)}
      />
      <Button onClick={_onClick} isWorking={isOpen} {...reactProps}>
        {props.children || <>Sign In</>}
      </Button>
    </WagmiConfig>
  )
}
export default SignInButton
