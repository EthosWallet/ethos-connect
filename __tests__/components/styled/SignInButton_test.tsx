import React from 'react'
import { create, act } from 'react-test-renderer'

import SignInButton from '../../../src/components/styled/SignInButton'
import SignInModal from '../../../src/components/styled/SignInModal'
import Button from '../../../src/components/headless/Button'

describe('SignInButton', () => {
  it('renders correctly', () => {
    const rendered = create(<SignInButton />).toJSON()
    expect(rendered).toMatchSnapshot()
  })

  it('triggers the SignInModal when clicked and triggers onClick callback', () => {
    const onClick = jest.fn()
    const signInButton = create(<SignInButton onClick={onClick} />)
    const root = signInButton.root
    expect(root.findAllByProps({ isOpen: true }).length).toBe(0)
    expect(onClick.mock.calls.length).toBe(0)

    act(() => {
      root.findByType(Button).props.onClick()
    })

    expect(root.findAllByProps({ isOpen: true }).length).toBe(1)
    expect(onClick.mock.calls.length).toBe(1)
  })
})
