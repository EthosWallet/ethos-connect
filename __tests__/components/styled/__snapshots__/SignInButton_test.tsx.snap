// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignInButton renders correctly 1`] = `
Array [
  <div
    role="dialog"
    style={
      Object {
        "display": "none",
        "position": "relative",
        "zIndex": "10",
      }
    }
  >
    <div
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "rgb(107 114 128 / .75)",
          "bottom": "0px",
          "left": "0px",
          "position": "fixed",
          "right": "0px",
          "top": "0px",
        }
      }
    />
    <div
      style={
        Object {
          "bottom": "0px",
          "left": "0px",
          "overflowY": "auto",
          "position": "fixed",
          "right": "0px",
          "top": "0px",
          "zIndex": "99",
        }
      }
    >
      <div
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "justifyContent": "center",
            "minHeight": "100%",
            "padding": "0",
            "textAlign": "center",
          }
        }
      >
        <div
          style={
            Object {
              "backgroundColor": "#ffffff",
              "borderRadius": "0.5rem",
              "boxShadow": "0 20px 25px -5px rgba(0, 0, 0, 0.1), 0 10px 10px -5px rgba(0, 0, 0, 0.04)",
              "marginBottom": "2rem",
              "marginTop": "2rem",
              "maxWidth": "40rem",
              "overflow": "hidden",
              "position": "relative",
              "textAlign": "left",
              "transitionProperty": "all",
              "width": "100%",
            }
          }
        >
          <div>
            <div
              style={
                Object {
                  "borderBottom": "1px solid rgb(241 245 249)",
                  "display": "flex",
                  "justifyContent": "space-between",
                  "padding": "12px",
                }
              }
            >
              <h3
                style={
                  Object {
                    "fontSize": "1rem",
                    "fontWeight": "500",
                    "margin": "0",
                  }
                }
              >
                Sign In
              </h3>
              <div
                onClick={[Function]}
                style={
                  Object {
                    "backgroundColor": "#F9FAFB",
                    "borderRadius": "100%",
                    "color": "#A0AEBA",
                    "cursor": "pointer",
                    "height": "24px",
                    "textAlign": "center",
                    "width": "24px",
                  }
                }
              >
                âœ•
              </div>
            </div>
            <div
              style={
                Object {
                  "display": "flex",
                  "justifyContent": "space-between",
                }
              }
            >
              <div
                style={
                  Object {
                    "borderBottom": "1px solid rgb(241 245 249)",
                    "borderRight": "1px solid rgb(241 245 249)",
                    "display": "flex",
                    "flexDirection": "column",
                    "gap": "6px",
                    "padding": "24px 12px",
                    "width": "300px",
                  }
                }
              >
                <div
                  onClick={[Function]}
                  style={
                    Object {
                      "backgroundColor": "#F9FAFB",
                      "borderRadius": "0.5rem",
                      "cursor": "pointer",
                      "fontWeight": "500",
                      "padding": "12px",
                    }
                  }
                >
                  <button
                    style={
                      Object {
                        "alignItems": "center",
                        "background": "none",
                        "border": "none",
                        "display": "flex",
                        "gap": "0.5rem",
                        "justifyContent": "start",
                        "textDecoration": "none",
                      }
                    }
                  >
                    <svg
                      fill="none"
                      height={23.51063829787234}
                      viewBox="0 0 47 65"
                      width={17}
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M6.00471 1H40.0029C42.7644 1 45.0029 3.23858 45.0029 6V44.8425C45.0029 47.604 42.7643 49.8425 40.0029 49.8425H6.0047C3.24328 49.8425 1.0047 47.604 1.0047 44.8425V6C1.0047 3.23858 3.24329 1 6.00471 1Z"
                        stroke="#1e293b"
                        strokeWidth="2"
                      />
                      <path
                        d="M6.68764 3.64648L30.6631 14.8026C32.0736 15.4589 32.9756 16.8735 32.9756 18.4292V58.6799C32.9756 61.5743 29.9966 63.5105 27.3515 62.3353L3.37601 51.683C1.93126 51.0411 1.00013 49.6085 1.00013 48.0276V7.27309C1.00013 4.34854 4.03609 2.41268 6.68764 3.64648Z"
                        fill="#1e293b"
                      />
                    </svg>
                    Ethos
                  </button>
                </div>
                <div
                  onClick={[Function]}
                  style={
                    Object {
                      "backgroundColor": "#F9FAFB",
                      "borderRadius": "0.5rem",
                      "cursor": "pointer",
                      "fontWeight": "500",
                      "padding": "12px",
                    }
                  }
                >
                  <button
                    style={
                      Object {
                        "alignItems": "center",
                        "background": "none",
                        "border": "none",
                        "display": "flex",
                        "gap": "0.5rem",
                        "justifyContent": "start",
                        "textDecoration": "none",
                      }
                    }
                  >
                    <svg
                      height={21.42857142857143}
                      id="SuiSvg"
                      viewBox="-1 0 28 40"
                      width={15}
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M1.8611,33.0541a13.6477,13.6477,0,0,0,23.7778,0,13.89,13.89,0,0,0,0-13.8909L15.1824.8368a1.6444,1.6444,0,0,0-2.8648,0L1.8611,19.1632A13.89,13.89,0,0,0,1.8611,33.0541ZM10.8044,9.5555,13.0338,5.648a.8222.8222,0,0,1,1.4324,0L23.043,20.68a10.8426,10.8426,0,0,1,.8873,8.8828,9.4254,9.4254,0,0,0-.4388-1.4586c-1.1847-3.0254-3.8634-5.36-7.9633-6.9393-2.8187-1.0819-4.618-2.6731-5.3491-4.73C9.2375,13.7848,10.221,10.8942,10.8044,9.5555ZM7.0028,16.2184,4.457,20.68a10.8569,10.8569,0,0,0,0,10.8582,10.6776,10.6776,0,0,0,16.1566,2.935,7.5061,7.5061,0,0,0,.0667-5.2913c-.87-2.1858-2.9646-3.9308-6.2252-5.1876-3.6857-1.4147-6.08-3.6233-7.1157-6.5625A9.297,9.297,0,0,1,7.0028,16.2184Z"
                        style={
                          Object {
                            "fill": "#6EBCEE",
                            "fillRule": "evenodd",
                          }
                        }
                      />
                    </svg>
                    Sui Test Wallet
                  </button>
                </div>
              </div>
              <div
                style={
                  Object {
                    "display": "flex",
                    "flexDirection": "column",
                    "gap": "12px",
                    "padding": "24px",
                  }
                }
              >
                <div>
                  <h3
                    style={
                      Object {
                        "fontWeight": "500",
                        "margin": "0",
                      }
                    }
                  >
                    Sign up or log in with Google
                  </h3>
                  <div
                    onClick={[Function]}
                    style={
                      Object {
                        "display": "flex",
                        "gap": "6px",
                        "padding": "12px 0",
                      }
                    }
                  >
                    <div
                      style={
                        Object {
                          "cursor": "pointer",
                        }
                      }
                    >
                      <svg
                        height={36}
                        viewBox="0 0 24 24"
                        width={36}
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <g
                          transform="matrix(1, 0, 0, 1, 27.009001, -39.238998)"
                        >
                          <path
                            d="M -3.264 51.509 C -3.264 50.719 -3.334 49.969 -3.454 49.239 L -14.754 49.239 L -14.754 53.749 L -8.284 53.749 C -8.574 55.229 -9.424 56.479 -10.684 57.329 L -10.684 60.329 L -6.824 60.329 C -4.564 58.239 -3.264 55.159 -3.264 51.509 Z"
                            fill="#4285F4"
                          />
                          <path
                            d="M -14.754 63.239 C -11.514 63.239 -8.804 62.159 -6.824 60.329 L -10.684 57.329 C -11.764 58.049 -13.134 58.489 -14.754 58.489 C -17.884 58.489 -20.534 56.379 -21.484 53.529 L -25.464 53.529 L -25.464 56.619 C -23.494 60.539 -19.444 63.239 -14.754 63.239 Z"
                            fill="#34A853"
                          />
                          <path
                            d="M -21.484 53.529 C -21.734 52.809 -21.864 52.039 -21.864 51.239 C -21.864 50.439 -21.724 49.669 -21.484 48.949 L -21.484 45.859 L -25.464 45.859 C -26.284 47.479 -26.754 49.299 -26.754 51.239 C -26.754 53.179 -26.284 54.999 -25.464 56.619 L -21.484 53.529 Z"
                            fill="#FBBC05"
                          />
                          <path
                            d="M -14.754 43.989 C -12.984 43.989 -11.404 44.599 -10.154 45.789 L -6.734 42.369 C -8.804 40.429 -11.514 39.239 -14.754 39.239 C -19.444 39.239 -23.494 41.939 -25.464 45.859 L -21.484 48.949 C -20.534 46.099 -17.884 43.989 -14.754 43.989 Z"
                            fill="#EA4335"
                          />
                        </g>
                      </svg>
                    </div>
                  </div>
                </div>
                <h3
                  style={
                    Object {
                      "fontWeight": "500",
                      "margin": "0",
                    }
                  }
                >
                  Or sign up or log in with a link
                </h3>
                <form
                  onSubmit={[Function]}
                >
                  <input
                    onChange={[Function]}
                    placeholder="Email address"
                    style={
                      Object {
                        "border": "1px solid rgb(203 213 225)",
                        "borderRadius": "0.5rem",
                        "padding": "6px",
                        "width": "90%",
                      }
                    }
                    type="email"
                    value=""
                  />
                  <button
                    style={
                      Object {
                        "backgroundColor": "#761AC7",
                        "border": "1px solid rgb(203 213 225)",
                        "borderRadius": "0.5rem",
                        "color": "#FFFFFF",
                        "marginTop": "1rem",
                        "minWidth": "6rem",
                        "padding": "6px",
                        "textDecoration": "none",
                      }
                    }
                    type="submit"
                  >
                    Send
                  </button>
                </form>
                <div
                  style={
                    Object {
                      "fontSize": "small",
                      "paddingLeft": "3px",
                      "paddingTop": "6px",
                    }
                  }
                >
                  Advanced:Â 
                  <a
                    href="test/self-custodial"
                    rel="noopener noreferrer"
                    style={
                      Object {
                        "color": "#751ac7",
                        "fontWeight": 400,
                        "textDecoration": "underline",
                      }
                    }
                    target="_blank"
                  >
                    Create A Self-Custodial Wallet
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <button
    onClick={[Function]}
  >
    Sign In
  </button>,
]
`;
